staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/lastappliedupdater.go

```diff
@@ -61,7 +63,11 @@ func (f *lastAppliedUpdater) Apply(liveObj, newObj runtime.Object, managed Manag
 		if err != nil {
 			return nil, nil, fmt.Errorf("failed to build last-applied annotation: %v", err)
 		}
-		err = setLastApplied(liveObj, lastAppliedValue)
+		if (int64)(len(lastAppliedValue)) >= totalAnnotationSizeLimitB {
```

Comment:
even if this single annotation is not over the limit, it can bump the total annotation size over the limit

see the validation done by https://github.com/kubernetes/kubernetes/blob/c24f2f8225be0aaf6f7f4743d7fc283040cfbf0d/staging/src/k8s.io/apimachinery/pkg/api/validation/objectmeta.go#L46-L60

Reply:
we need unit tests with an existing object with a last-applied annotation and other annotations *almost* at the limit:

1. an update via server-side apply that takes the total annotation size over the limit because of the new last-applied annotation (old object is ok, but old object + new last-applied is too big)
2. an update via server-side apply with additional annotations that takes the total annotation size over the limit because of the new annotations and new last-applied annotation (old object + new annotations is ok,  old object + new last-applied is ok, but old object + new annotations + new last-applied is too big)

Reply:
Okay, I think I've addressed these by checking the entire annotations size limit as we do in validation and adding the appropriate unit tests.